python/pyspark/core/context.py

```diff
@@ -554,6 +554,28 @@ def setSystemProperty(cls, key: str, value: str) -> None:
         assert SparkContext._jvm is not None
         SparkContext._jvm.java.lang.System.setProperty(key, value)
 
+    @classmethod
+    def getSystemProperty(cls, key: str) -> str:
+        """
+        Get a Java system property, such as `java.home`.
+
+        .. versionadded:: 4.0.0
+
+        Parameters
+        ----------
+        key : str
+            The key of a new Java system property.
+
+        Examples
+        --------
+        >>> sc.getSystemProperty("SPARK_SUBMIT")
+        'true'
+        >>> _ = sc.getSystemProperty("java.home")
+        """
+        SparkContext._ensure_initialized()
+        assert SparkContext._jvm is not None
+        return SparkContext._jvm.java.lang.System.getProperty(key)
```

Comment:
three questions:
1, shall we name it to be more specific to the Java?
2, do we need to also introduce it in Scala side?
3, shall we make it a SparkSession method? so that we can enable it in SparkConnect later

also cc @cloud-fan for this new feature.

Reply:
To @zhengruifeng , as I wrote in the PR description, this is a symmetric version of the existing `setSystemProperty`. So, no for (1) and (3). For (2),  Scala can access `System.getProperty` directly already.

> https://spark.apache.org/docs/4.0.0-preview2/api/python/reference/api/pyspark.SparkContext.setSystemProperty.html